homeassistant/components/smappee/sensor.py

```diff
@@ -291,17 +313,34 @@ def device_class(self):
         """Return the class of this device, from component DEVICE_CLASSES."""
         return self._device_class
 
+    @property
+    def state_class(self):
+        """Return the state class of this device."""
+        return STATE_CLASS_MEASUREMENT
+
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
         return self._unit_of_measurement
 
+    @property
+    def last_reset(self):
+        """Return the time when the sensor was last reset, if any."""
+        if self._sensor in ("power_today", "solar_today", "alwayson_today", "switch"):
+            return start_of_local_day()
+        elif self._sensor in ("power_current_hour", "solar_current_hour"):
+            pass
+        elif self._sensor in ("power_last_5_minutes"):
+            pass
```

Comment:
No need to add this, if it is being passed on.
```suggestion
```

Reply:
I'm looking for a start_of_local_hour helper method

Reply:
I'm not sure, but the description says: Power, not Energy, hence it made me think it is not metered value, thus should not have a last_reset?

Reply:
no, those are actual metered values (underlying names are not well chosen)

Reply:
I've made an attempt by using the utcnow helper and replacing the minutes value